tools/tests/test_pretty_print.py

```diff
@@ -0,0 +1,191 @@
+from __future__ import absolute_import
+from __future__ import print_function
+
+import unittest
+
+from tools.lib.pretty_print import pretty_print_html
+
+# Note that GOOD_HTML isn't necessarily beautiful HTML.  Apart
+# from adjusting indentation, we mostly leave things alone to
+# respect whatever line-wrapping styles were in place before.
+
+BAD_HTML = """
+<!-- test -->
+<!DOCTYPE html>
+
+
+
+<html>
+    <!-- test -->
+    <head>
+        <title>Test</title>
+        <meta charset="utf-8" />
+        <link rel="stylesheet" href="style.css" />
+    </head>
+    <body>
+      <div><p>Hello<br />world!</p></div>
+        <p>Goodbye<!-- test -->world!</p>
+        <table>
+           <tr>
+                       <td>5</td>
+           </tr>
+        </table>
+    <pre>
+            print 'hello world'
+    </pre>
+         <div class = "foo"
+              id = "bar"
+              role = "whatever">{{ bla }}</div>
+    </body>
+</html>
+<!-- test -->
+"""
+
+GOOD_HTML = """
+<!-- test -->
+<!DOCTYPE html>
+
+
+
+<html>
+    <!-- test -->
+    <head>
+        <title>Test</title>
+        <meta charset="utf-8" />
+        <link rel="stylesheet" href="style.css" />
+    </head>
+    <body>
+        <div><p>Hello<br />world!</p></div>
+        <p>Goodbye<!-- test -->world!</p>
+        <table>
+            <tr>
+                <td>5</td>
+            </tr>
+        </table>
+        <pre>
+                print 'hello world'
+        </pre>
+        <div class = "foo"
+            id = "bar"
+             role = "whatever">{{ bla }}</div>
```

Comment:
@tommyip We want the beautifier to be not overly aggressive about fixing every little quirk of the HTML, in case the original coders had some reason to format things the way we did.  That decision may change, though; we might want to move toward a more opinionated formatter.